src/rpm/package.rs

```diff
@@ -324,9 +326,10 @@ impl RPMPackageMetadata {
     }
 
     #[inline]
-    pub fn get_payload_compressor(&self) -> Result<&str, RPMError> {
-        self.header
-            .get_entry_data_as_string(IndexTag::RPMTAG_PAYLOADCOMPRESSOR)
+    pub fn get_payload_compressor(&self) -> Result<Compressor, RPMError> {
+        let comp_str = self.header
+            .get_entry_data_as_string(IndexTag::RPMTAG_PAYLOADCOMPRESSOR)?;
+        Compressor::from_str(comp_str)
```

Comment:
I'm not sure about this as it would be an exception from how it works for all other accessors, we're just returning the values in a mostly-raw form.  I understand the benefit of parsing to an enum type in many cases but I'm not so sure it applies strongly here?

Open to discussion on this point

Reply:
@dralley hello! Thank you for the review! I have some thoughts:

1. `mostly-raw` is `bytes` for me, but here is `&str`. We did some simple parsing before.
2. `crate` doesn't have API for working with this strings. I can only print information about Package but I cannot work with it.
3. I think `enum Compress` can have `Compress::Unknow(String)`  were we can store unknowing compress methods and didn't raise error

Reply:
I prefer the enum, if only a limited set of strings are used/supported and raise an error otherwise.

Reply:
Removed

Reply:
So, as currently implemented it looks like this returns not a compression type, but a `Compressor` instance, which feels like it would be kind of awkward to match against, because as a user I'm never going to use it as a compressor and I don't care about the associated data, just the tags.

Reply:
I'd tend to say yes, but I think anything typed is better than strings.

Reply:
>It's kinda nuts that we expose the current Compressor type publicly for the builder pattern.

Well not entirely as you might want to e.g. tweak the compression level, and we provide no other way to do that.  But long term I agree it's not ideal.

Reply:
Well, ok. I agree with you. I will push changes today or tomorrow)

Reply:
@ikrivosheev Reminder :)

Reply:
@dralley I remember! Sorry for the long answer!

Reply:
Done